File: cpp/Mahasiswa.cpp
#include <iostream>
#include <string>

using namespace std;

//deklarasi kelas mahasiswa
class Mahasiswa
{

	//atribut untuk kelas mahasiswa
	private:
		//semua atribut bertipe string
		string nim;
		string nama;
		string gender;
		string prodi;
		string fakultas;

	public:

		//constructor
		Mahasiswa(){
			this->nim = "";
			this->nama = "";
			this->gender = "";
			this->prodi = "";
			this->fakultas = "";
		}

		//constructor dengan parameter berupa atributnya
		Mahasiswa(string nim, string nama, string gender, string prodi, string fakultas){
			this->nim = nim;
			this->nama = nama;
			this->gender = gender;
			this->prodi = prodi;
			this->fakultas = fakultas;
		}


		//setter untuk tiap atribut (untuk mengubah value atribut)

		void set_nim(string nim){
			this->nim = nim;
		}

		void set_nama(string nama){
			this->nama = nama;
		}

		void set_gender(string gender){
			this->gender = gender;
		}

		void set_prodi(string prodi){
			this->prodi = prodi;
		}

		void set_fakultas(string fakultas){
			this->fakultas = fakultas;
		}


		//getter untuk tiap atribut (untuk mendapatkan value atribut)
		string get_nim(){
			return this->nim;
		}

		string get_nama(){
			return this->nama;
		}

		string get_gender(){
			return this->gender;
		}

		string get_prodi(){
			return this->prodi;
		}

		string get_fakultas(){
			return this->fakultas;
		}

		//destructor
		~Mahasiswa(){

		}

};

File: python/Mahasiswa.py
class Mahasiswa:

	__nim = ""
	__nama = ""
	__gender = ""
	__prodi = ""
	__fakultas = ""

	def __init__(self, nim = "", nama = "", gender = "", prodi = "", fakultas = ""):
		self.__nim = nim
		self.__nama = nama
		self.__gender = gender
		self.__prodi = prodi
		self.__fakultas = fakultas

	#setter

	def set_nim(self, nim):
		self.__nim = nim;

	def set_nama(self, nama):
		self.__nama = nama;

	def set_gender(self, gender):
		self.__gender = gender;

	def set_prodi(self, prodi):
		self.__prodi = prodi;

	def set_fakultas(self, fakultas):
		self.__fakultas = fakultas;

	#getter

	def get_nim(self):
		return self.__nim
	
	def get_nama(self):
		return self.__nama
	
	def get_gender(self):
		return self.__gender
	
	def get_prodi(self):
		return self.__prodi
	
	def get_fakultas(self):
		return self.__fakultas
	